lib/reek/tree_walker.rb

```diff
@@ -53,10 +54,6 @@ def walk
     # RootContext -> children: 1 ModuleContext -> children: 1 MethodContext
     #
     # @return [Reek::Context::RootContext] tree of nested contexts
-    def context_tree
-      @context_tree ||= process(exp)
```

Comment:
I may be mostly to blame for this overuse, as I tend to favour memoization over initialization in the constructor. In this case I don't see a problem moving it since generally you will want to call `#walk` on the TreeWalker anyway.